### PR TITLE
Report usage analytics.

### PR DESCRIPTION
**Pull request by ira-gordin-sap**

### CODE REVIEW

File: backend/src/swa-tracker/swa-tracker-wrapper.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	EXPLORE_AND_INSTALL_GENERATORS_LINK: "Explore and Install Generators link",
+	PREVIOUS_STEP: "One of the previous steps is clicked"
+}; 
+
+/**
+ * A Simple Wrapper to hold the state of our "singleton" (per extension) SWATracker
+ * implementation.
+ */
+
+export const ERROR_ANALYTICS_TRACKER_NOT_INITIALIZED = 'Analytics tracker has not yet been initialized!';
```
Comment (slavik-lvovsky) [inline backend/src/swa-tracker/swa-tracker-wrapper.ts:21]: should not be exported

File: backend/src/yeomanui.ts [hunk 2]
```diff
@@ -50,6 +51,7 @@ export class YeomanUI {
   private readonly replayUtils: ReplayUtils;
   private readonly customQuestionEventHandlers: Map<string, Map<string, Function>>;
   private errorThrown = false;
+  private startTime: number;
```
Comment (slavik-lvovsky) [inline backend/src/yeomanui.ts:54]: i think that you do not need this param for usage analitics

**Review by slavik-lvovsky — CHANGES_REQUESTED** [review]

File: backend/src/yeomanui.ts (hunk 2 above)
Comment (ira-gordin-sap) [reply backend/src/yeomanui.ts:54]: I need - Avital wants to know how much time takes to project to be generated.

Comment (ira-gordin-sap) [reply backend/src/yeomanui.ts]: Done.

**Review by slavik-lvovsky — APPROVED** [review]

File: backend/src/swa-tracker/swa-tracker-wrapper.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	getSWA().track(eventType);
+	getLogger().trace("SAP Web Analytics tracker was called", {eventType});
+}
+
+export function updateOneOfPreviousStepsClicked(generatorName: string) {
+	const eventType = EVENT_TYPES.PREVIOUS_STEP;
+	let customEvents = [generatorName];
+	getSWA().track(eventType, customEvents);
+	getLogger().trace("SAP Web Analytics tracker was called", {eventType, generatorName, customEvents});  
+}
```
Comment (tomer-epstein) [inline backend/src/swa-tracker/swa-tracker-wrapper.ts:108]: looks ok to me too.
 refactor I would do is:
export from this file only 1 class called SWA which contains all the funcs.

File: backend/src/yeomanui.ts [hunk 4]
```diff
@@ -17,6 +17,7 @@ import Generator = require("yeoman-generator");
 import { GeneratorFilter, GeneratorType } from "./filter";
 import { IChildLogger } from "@vscode-logging/logger";
 import {IPrompt} from "@sap-devx/yeoman-ui-types";
+import { updateGeneratorStarted, updateGeneratorEnded, updateExploreAndInstallGeneratorsLinkClicked, updateOneOfPreviousStepsClicked } from "./swa-tracker/swa-tracker-wrapper";
```
Comment (tomer-epstein) [inline backend/src/yeomanui.ts:20]: import { SWA } from "./swa-tracker/swa-tracker-wrapper";

**Pull request merged**